File: Backend/Travle.Model/Responses/DestinationCategoryResponse.cs
namespace Travle.Model.Responses
{
    public class DestinationCategoryResponse
    {
        public int Id { get; set; }
        public string Name { get; set; } = string.Empty;

        /// <summary>Short blurb shown on the onboarding category cards; null when unset.</summary>
        public string? Description { get; set; }

        /// <summary>Small PNG thumbnail for the onboarding grid / admin list. The full image is served by the
        /// dedicated <c>GET /DestinationCategories/{id}/image</c> endpoint, never on this list DTO (§12).</summary>
        public byte[]? ImageThumbnail { get; set; }

        public DateTime CreatedAt { get; set; }
        public DateTime? ModifiedAt { get; set; }
    }
}

File: UI/travle_core/lib/src/models/destination_category_response.dart
import 'package:json_annotation/json_annotation.dart';

part 'destination_category_response.g.dart';

/// A destination category (mirrors the backend `DestinationCategoryResponse`).
/// Lists carry only [imageThumbnail] — a small base64 PNG shown on the onboarding
/// cards; the full illustration is served by `GET /DestinationCategories/{id}/image`.
@JsonSerializable()
class DestinationCategoryResponse {
  DestinationCategoryResponse({
    required this.id,
    required this.name,
    this.description,
    this.imageThumbnail,
    required this.createdAt,
    this.usageCount = 0,
    this.deleteBlockedReason,
    this.modifiedAt,
  });

  final int id;
  final String name;

  /// Short blurb shown on the onboarding category card; null when unset.
  final String? description;

  /// Small base64 PNG thumbnail for the onboarding grid / admin list.
  final String? imageThumbnail;

  /// How many other records still reference this row (0 = deletable).
  @JsonKey(defaultValue: 0)
  final int usageCount;

  /// Why this row cannot be deleted, or null when it can — rendered as the
  /// disabled Delete button's tooltip (course UI rule: unavailable actions are
  /// disabled with the reason shown).
  final String? deleteBlockedReason;

  final DateTime createdAt;
  final DateTime? modifiedAt;

  factory DestinationCategoryResponse.fromJson(Map<String, dynamic> json) =>
      _$DestinationCategoryResponseFromJson(json);

  Map<String, dynamic> toJson() => _$DestinationCategoryResponseToJson(this);
}
